src/core/agent/services/connectionService.ts

```diff
@@ -388,16 +391,7 @@ class ConnectionService extends AgentService {
         type: EventTypes.OperationAdded,
         payload: { operation: pendingOperation },
       });
-    } else if (!operation.done) {
-      throw new Error(ConnectionService.FAILED_TO_RESOLVE_OOBI);
```

Comment:
We need to throw in if it's not complete in the `if (waitForCompletion) {` block